From 301d7fd539ab26fd3c4ee83bb51478578a081a94 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 3 Jul 2025 11:35:31 +0300
Subject: [PATCH 1/1] [utilities] Nokia7215:WA: no SAI start on config-reload

https://github.com/sonic-net/sonic-buildimage/issues/22994
Config reload command doesn't start SAI on branch master/202505 on board Nokia-7215/AC3
Same/similar is for load minigraph command

Current WORKAROUND:
add "restart swss" at the end of "reload" command handling in file

[src/sonic-utilities/]config/main.py
On board: /usr/local/lib/python3.11/dist-packages/config/main.py

======================================================================
Description of the bug

Marvell SAI starting is never achieved on command "config reload -y".
Last worked - branch 202411, stopped to work properly on branch master and branch 202505.
Platform: marvell-prestera, ARCH=armhf, board Nokia-7215(AC3)

Problem suspect -- orchestration synchronization on STOP service.

Observations are:
- Both swss and syncd containers are Up/Running but SAI is not achieved
   not started (according to syslog).
- apply "systemctl restart swss" fixes the problem
- apply "systemctl restart sonic.target"
   (which is last part of "config reload" implementation) is always FAIL
- Manual Stop sequence (with sleeps) emulating "stop sonic.target"
   followed by "start sonic.target" is OK!
- The "config reload" has also DB-config and migration processing which
   also cause problem

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 config/main.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/config/main.py b/config/main.py
index 0a5b1e32..6338cd04 100644
--- a/config/main.py
+++ b/config/main.py
@@ -1043,6 +1043,12 @@ def _restart_services():
     click.echo("Reloading Monit configuration ...")
     clicommon.run_command(['sudo', 'monit', 'reload'])
 
+    # ARMHF/Nokia-7215: SAI never started by syncd. WA force restart swss (with syncd)
+    time.sleep(10)
+    clicommon.run_command(['sudo', 'systemctl', 'stop', 'swss'])
+    clicommon.run_command(['sudo', 'systemctl', 'stop', 'syncd'])
+    clicommon.run_command(['sudo', 'systemctl', 'restart', 'swss'])
+
 def _per_namespace_swss_ready(service_name):
     out, _ = clicommon.run_command(['systemctl', 'show', str(service_name), '--property', 'ActiveState', '--value'], return_cmd=True)
     if out.strip() != "active":
-- 
2.25.1

